src/cascadia/TerminalControl/TermControl.h

```diff
@@ -183,6 +183,12 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
         winrt::Windows::UI::Xaml::Controls::SwapChainPanel::LayoutUpdated_revoker _layoutUpdatedRevoker;
 
+        inline bool _IsClosing() const noexcept {
+            // _closing isn't atomic and may only be accessed from the main thread.
+            assert(Dispatcher().HasThreadAccess());

```

Comment:
I previously wanted to introduce a cool macro for this in WinRTUtils, but I realized, that global state would only work properly if it's shared using a .dll. A .lib like that wouldn't do that for us. 😟

Reply:
This assert is compiled out in debug builds, correct?

Reply:
Sorry, compiled out in release builds./